gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -171,12 +191,15 @@ public void shouldRedirectToMFAEnrollmentPage_adaptiveMFA() throws Exception {
     @Test
     public void shouldRedirectToMFAEnrollmentPage_adaptiveMFA_no_enroll() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String rule = "{context.attributes['geoip']['country_iso_code'] == 'FR'";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.REQUIRED, rule);
```

Comment:
**todo**: the test is named shouldRedirectToMFAEnrollmentPage_**adaptiveMFA**_no_enroll but the ChallengeType is set to REQUIRED, should be CONDITIONAL